sklearn/preprocessing/data.py

```diff
@@ -1946,3 +1957,149 @@ def transform(self, X):
         """
         return _transform_selected(X, self._transform,
                                    self.categorical_features, copy=True)
+
+
+def _boxcox(x, lambda_x=None):
+    if lambda_x is None:
+        x, lambda_x = stats.boxcox(x, lambda_x)
+        return x, lambda_x
+    else:
+        x = stats.boxcox(x, lambda_x)
+        return x
+
+
+def boxcox(X, copy=True):
+    """BoxCox transform to the input data
+
+    Apply boxcox transform on individual features with lambda
+    that maximizes the log-likelihood function for each feature
+
+    Parameters
+    ----------
+    X : array-like, shape (n_samples, n_features)
+        The data to be transformed. Should contain only positive data.
+
+    copy : boolean, optional, default is True
+        set to False to perform inplace transformation and avoid a
+        copy (if the input is already a numpy array or a scipy.sparse
+        CSR matrix and if axis is 1).
+
+    Returns
+    -------
+    X_tr : array-like, shape (n_samples, n_features)
+        The transformed data.
+
+    References
+    ----------
+    G.E.P. Box and D.R. Cox, "An Analysis of Transformations", Journal of the
+    Royal Statistical Society B, 26, 211-252 (1964).
+    """
+    X = check_array(X, ensure_2d=True, dtype=FLOAT_DTYPES, copy=copy)
+    if any(np.any(X <= 0, axis=0)):
+        raise ValueError("BoxCox transform can only be applied "
+                         "on positive data")
+    n_features = X.shape[1]
+    outputs = Parallel(n_jobs=-1)(delayed(_boxcox)(X[:, i], lambda_x=None)
+                                  for i in range(n_features))
+    output = np.concatenate([o[0][..., np.newaxis] for o in outputs], axis=1)
+    return output
+
+
+class BoxCoxTransformer(BaseEstimator, TransformerMixin):
+    """BoxCox transformation on individual features.
+
+    Boxcox transform wil be applied on each feature (each column of
+    the data matrix) with lambda evaluated to maximise the log-likelihood
+
+    Parameters
+    ----------
+    feature_indices : None | array of int or mask, default None
```

Comment:
Copy the semantics of `categorical_features` used elsewhere. 'all' is the default, i.e. a string. I would not call it `feature_indices`, but `features` or `transformed_features` or `exponential_features` or `which_features`